### PR TITLE
Move to AVA for testing

### PR DESCRIPTION
**Pull request by tdeschryver**
With this change all tests are being run with `AVA` instead of `mocha`.

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
+/* eslint-env mocha */
+'use strict';
+const fs = require('fs');
+const path = require('path');
+const test = require('ava');
+const stringifyObject = require('../');
```
Comment (sindresorhus) [inline test/index.js]: You can use `import` syntax here as test files are transpiled with Babel automatically. You can also drop line 1 and 2.

Example: https://github.com/sindresorhus/indent-string/blob/458eca3f626b95bdcff5afe30d1568bf76889920/test.js#L1

Comment (kevva) [inline test/index.js]: Use `module` syntax here (`ava` transpiles tests with `babel` automatically).

```js
import fs from 'fs';
...
```

File: test/index.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
+/* eslint-env mocha */
```
Comment (kevva) [inline test/index.js]: Remove this.

File: test/index.js [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+		arr: [1, 2, 3],
+		nested: {hello: 'world'}
+	};
+
+	t.is(stringifyObject(object), '{\n\tarr: [\n\t\t1,\n\t\t2,\n\t\t3\n\t],\n\tnested: {\n\t\thello: \'world\'\n\t}\n}');
+	t.is(stringifyObject(object, {inlineCharacterLimit: 12}), '{\n\tarr: [1, 2, 3],\n\tnested: {\n\t\thello: \'world\'\n\t}\n}');
+});
+
+test('handles non-plain object', t => {
+	t.notDeepEqual(stringifyObject(fs.statSync(__filename)), '[object Object]');
```
Comment (kevva) [inline test/index.js]: `t.not` since it's a `string`.

**Comment by tdeschryver** [comment]
Any ideas about the failing test on node 4? :confused:

**Comment by kevva** [comment]
> Any ideas on the failing test on nodejs 4? 😕

Because the test file is transpiled by Babel. You could make it a named `function` instead:

```js
fn: function () {}
```

And then in `object.js`:

```
fn: function fn() {}
```

**Comment by tdeschryver** [comment]
@kevva thanks for pointing that out, would have taken me a while to figure it out...
I've had to change it in both places to `fn: function fn() {}`.
Otherwise the test was failing:
image

**Comment by tdeschryver** [comment]
@kevva I just saw your edit. Is my way of doing it OK?

**Comment by kevva** [comment]
> [@kevva](https://github.com/kevva) I just saw your edit. Is my way of doing it OK?

Yup, it's fine :).

**Comment by sindresorhus** [comment]
Nice work @tdeschryver! Thanks for doing it :)

**Pull request merged**